common/db.go

```diff
@@ -156,6 +159,18 @@ var schema = `
 	);
 	-- Index to only retrieve unbonding locks that have not been used
 	CREATE INDEX IF NOT EXISTS idx_unbondinglocks_usedblock ON unbondingLocks(usedBlock);
+
+	CREATE TABLE IF NOT EXISTS winningTickets (
+		sender STRING,
+		recipient STRING,
+		faceValue BLOB,
+		winProb BLOB,
+		senderNonce INTEGER,
+		recipientRand BLOB,
+		recipientRandHash STRING,
+		sig BLOB,
+		sessionID STRING
```

Comment:
Generally I like to incorporate a creation timestamp here as well. Useful for forensic purposes and doesn't affect any queries with a `DEFAULT CURRENT_TIMESTAMP`. Also, some constraints might have been nice such as a composite key consisting of`(recipientRand, senderNonce)` which would have ensured uniqueness of the pair.

Haven't fully read through the approach here yet, but I've used DB constraints as one form of checking to ensure subsequent code-paths are only executed once (eg, preventing double-submits to the blockchain).

Reply:
I'm more than happy to add a creation timestamp! Created this issue https://github.com/livepeer/go-livepeer/issues/661.

As for the constraints - the reason I did not add them is because they weren't necessary as of yet, and I don't like adding "future code" that isn't being used when it's added. I really can't see in what scenario we would benefit from this constraint since I don't see a code path that would lead to our saving the same winning ticket twice (since in-memory protections are in place to reject a repeat incoming ticket). So on this one I'm still leaning more towards not adding anything that isn't really needed, but would love to hear your further thoughts to learn more about why you think it's good to add?

As for double-submit prevention - I agree it's a good idea to defend against that, and think we can do that well by adding some ETH-related columns when we get to spending more time on a robust redemption mechanism. I think we will add these DB columns as we solve for this issue https://github.com/livepeer/go-livepeer/issues/632. wdyt?

Reply:
> why you think it's good to add?

The ability to declare constraints among fields and key relationships is probably my favorite feature of relational DBs. It's a type system that helps ensure the "shape" (consistency and integrity) of the data during storage and retrieval.

>  I really can't see in what scenario we would benefit from this constraint since I don't see a code path that would lead to our saving the same winning ticket twice (since in-memory protections are in place to reject a repeat incoming ticket)

We may have in-memory protections now, but any constraints act as a safety net in case these are missed somehow. Actually, I have been even lazier in the past: I've used the DB in lieu of in-memory protection, because why write additional code when your logic can be enforced declaratively?

Anyway, just my two cents! This is still fine as-is.